### DISCUSSION TITLE
Counting Blog Post 'Likes'

### DISCUSSION THREAD

**Discussion by mrpavlitzky** [OrchardCMS/OrchardCore#12249] [Q&A]
Hi,

I'm preparing to implement 'Thumbs Up' functionality for blog posts and I have a following dillema. Since I don't have much experience with document based databases (which as far as I know is how Orchard stores Content Items) I wonder what's the most optimal way to store the number of likes of every post. Suppose I'd like to add a module which shows the number of likes under each post. Would it then be reasonable to add a numeric field to the Blog Post content definition or maybe create a separate sql table with post_id and likes_count columns.

**Comment by hyzx86** [comment]
Enable the  `SQL Index` Feature

Define a ContentType like this , (not C# only，you can create with UI also):

``` C#
public class MyCollectionPart : ContentPart
{
   public ContentPickerField PostId {get; set;}
   public UserPickerField UserId {get; set;}
}

```

Then you can add a content to `MyCollection`,  after user click.

**Reply by mrpavlitzky** [reply]
@hyzx86 thanks for the suggestion. I'll have posts liked by unregistered users so I think I'll stick to updating a LikeCount content part on click event in every blog post.
